File: leetcode-solutions/0041-first-missing-positive/solution.cpp
class Solution {
public:
    int firstMissingPositive(vector<int>& nums) {
        sort(nums.begin(),nums.end());
        int i=0,ctr=1;
        while(i<nums.size()&&nums[i]<=0) i++;
        for(;i<nums.size();i++)
        {
            if(nums[i]>ctr)
            {
                return ctr;
            }
            else if(nums[i]==ctr)
            {
                ctr++;
            }
        }
        return ctr;
    }
};

File: leetcode-solutions/0041-first-missing-positive/solution.java
class Solution {
    public int firstMissingPositive(int[] nums) {
        Arrays.sort(nums);
        int num=1;
        for(int n:nums)
        {
            if(n==num)
            {
                num++;
                
            }
        }
        System.out.println(num);
        return num;
    }
}

